7506: Incorrect numeric formatting of PID by JMC

Suchitainf opened the pull request:
This PR is to fix the formatting issue of the JVM PID on JVM information screen. JVM PID attribute is received as NUMBER from JFR and hence the JMC application is treating is as number. When the number is smaller , the PID is displayed using commas and when the number is larger, its converted to exponential format. This is a long pending bug in product.

The PID (Process Id) attribute is present in JMC as part of System Process event also where the data type is correct i.e. String. So in order to maintain consistency, we have made the content type of PID attribute of JVM Information event also as String. We are converting the number value of PID to string and then storing it. 

We have also introduced a new aggregator PID which will contain the value of Identifier in plain text format instead of number format. This new aggregator is displayed on JVM Information page.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (1 review required, with at least 1 [Committer](https://openjdk.org/bylaws#committer))

### Issue
 * [JMC-7506](https://bugs.openjdk.org/browse/JMC-7506): Incorrect numeric formatting of PID by JMC (**Bug** - P3)


### Reviewers
 * [Brice Dutheil](https://openjdk.org/census#bdutheil) (@bric3 - Committer)
 * [Virag Purnam](https://openjdk.org/census#vpurnam) (@viragpurnam - Committer) ⚠️ Review applies to [9ed2439d](https://git.openjdk.org/jmc/pull/565/files/9ed2439d6906bac53c4ffbb587078b80897ff95a)
 * [Marcus Hirt](https://openjdk.org/census#hirt) (@thegreystone - **Reviewer**)


### Contributors
 * Marcus Hirt `<hirt@openjdk.org>`

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jmc.git pull/565/head:pull/565` \
`$ git checkout pull/565`

Update a local copy of the PR: \
`$ git checkout pull/565` \
`$ git pull https://git.openjdk.org/jmc.git pull/565/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 565`

View PR using the GUI difftool: \
`$ git pr show -t 565`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jmc/pull/565.diff">https://git.openjdk.org/jmc/pull/565.diff</a>

</details>


### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jmc/pull/565#issuecomment-2131475912)

viragpurnam reviewed (approved):
LGTM

thegreystone commented:
/contributor add thegreystone

thegreystone commented on core/org.openjdk.jmc.flightrecorder/src/main/java/org/openjdk/jmc/flightrecorder/jdk/JdkAggregators.java:
```diff
@@ -115,8 +115,7 @@ public final class JdkAggregators {
 
 	// VM Info
 	public static final IAggregator<String, ?> JVM_NAME = distinctAsString(VM_INFO, JdkAttributes.JVM_NAME);
-	public static final IAggregator<IQuantity, ?> JVM_PID = min(JdkAttributes.JVM_PID.getName(), null, VM_INFO,
```
Part of the public API. Perhaps better to deprecate?

Suchitainf replied:
Do you mean to use @deprecate tag here instead of deleting it completely?

bric3 replied:
Yeh otherwise, this would be a breaking change for consumer of the flight recorder libraries.

thegreystone replied:
Precisely. We can remove it in JMC 10. :)

Suchitainf commented:
/contributor add @thegreystone

Suchitainf commented:
/contributor add @thegreystone

Suchitainf commented:
I have added the JVM_PID aggregator and marked it as deprecated. In order to test it, I have added it on JVM Internal screen alongwith PID aggregator. Both the aggregators PID and JVM_PID got displayed like below:

<img width="687" alt="image" src="https://github.com/openjdk/jmc/assets/11155712/d1ad9197-2263-4894-8980-4d547b4eda04">

Note: Its just for the testing purpose. The second field which is in number format is not a part of the PR. Only the first field which is the proper format is the part of PR.

thegreystone commented:
Copyright checks failing.

Suchitainf commented:
/integrate
